src/api.rs

```diff
@@ -27,7 +27,203 @@ pub async fn world() -> Option<NamedFile> {
 
 #[get("/test")]
 pub async fn test() -> &'static str {
-    "Hello, test!"
+    println!(
+        "{:?}",
+        get_package(
+            "/home/albert/Documents/purdue/4sp/ECE461/module-registry/packages/postcss.zip",
+            "name"
+        )
+    );
+
+    "cool"
+}
+
+#[post("/package", data = "<package>")]
+pub async fn package_create(
+    package: Json<PackageData>,
+    mod_db: &State<ModuleDB>,
+) -> (Status, Either<Json<Package>, &'static str>) {
+    let mod_r = mod_db.read().await;
+
+    // if Content field is set
+    if package.Content != None && package.URL == None {
+        if let Ok(_) = base64_to_zip(
+            package.Content.as_ref().unwrap().as_str(),
+            "./temp_ece461.zip",
+        )
+        .await
+        {
+            let name = get_package("./temp_ece461.zip", "name");
+            let version = get_package("./temp_ece461.zip", "version");
+            let url = format!("https://www.npmjs.com/package/{:?}", name);
```

Comment:
`name` has type `Option<String>`, so I don't think you want it passed into the format function. Also be careful that `{:?}` prints strings with double quotes around them, so you probably want `{}` instead.